engine/battle/core.asm

```diff
@@ -6752,7 +6752,7 @@ HandleExplodingAnimation:
 	ld a, [wMoveMissed]
 	and a
 	ret nz
-	ld a, 5
+	ld a, MEGA_PUNCH
```

Comment:
The constant used for `wAnimationType` should not be a move constant. See the routine below:

https://github.com/pret/pokered/blob/b302e93674f376f2881cbd931a698345ad27bec3/engine/battle/animations.asm#L475-L498

`5` is associated with `ShakeScreenHorizontallyLight`.

So we should probably establish new constants; such as:

 ```
 	const_def 1
 	const ANIMATIONTYPE_SHAKE_SCREEN_VERTICALLY
 	const ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_HEAVY
 	const ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_SLOW
 	const ANIMATIONTYPE_BLINK_ENEMY_MON_SPRITE
 	const ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_LIGHT
 	const ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_SLOW_2
 ```
 
 and instead of your change you would do:
 
 ```
 ld a, ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_LIGHT
 ```
 
 Although I admit those animation type names are not the best; I think just adding constants right now is good enough for the PR.

Reply:
Ah okay the above comment applies.. but I missed the fall through to the wAnimationID on the following line.. let me think on this for a minute.

Reply:
In addition to the new constants.. What if we did something like this:

```diff
-	ld a, 5
+	ld a, ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_LIGHT
	ld [wAnimationType], a
+	assert ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_LIGHT == MEGA_PUNCH, \
+		"ANIMATIONTYPE_SHAKE_SCREEN_HORIZONTALLY_LIGHT does not equal MEGA_PUNCH, breaking assumption in HandleExplodingAnimation: fallthrough to wAnimationID"
+; fallthrough
PlayMoveAnimation:
	ld [wAnimationID], a
```

Reply:
I'm fine with those names unless/until we rename the relevant routine+wram labels. As for where to define those constants: would this be the time for a constants/battle_anim_constants.asm file?